go-controller/pkg/ovn/master_test.go

```diff
@@ -1678,6 +1679,167 @@ var _ = ginkgo.Describe("Default network controller operations", func() {
 		})
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
+
+	ginkgo.It("verify IPs in host-network-namespace address_set after remote zone node add", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := config.InitConfig(ctx, nil, nil)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			config.OVNKubernetesFeature.EnableInterconnect = true
+			config.Kubernetes.HostNetworkNamespace = "ovn-host-network"
+
+			// PortBinding is required to add interconect resources for
+			// remote zone nodes
+			pb := &sbdb.PortBinding{
+				LogicalPort: "tstor-node1",
+				Datapath:    "tstor-node1",
+				TunnelKey:   5,
+			}
+			dbSetup.SBData = append(
+				dbSetup.SBData,
+				pb,
+			)
+
+			// Set annotations required to create remote zone node
+			newNodeSubnet := "10.1.2.0/24"
+			transitswitchSubnet := "100.88.0.3/16"
+			testNode.Annotations["k8s.ovn.org/network-ids"] = fmt.Sprintf("{\"default\":\"%d\"}", 0)
+			testNode.Annotations["k8s.ovn.org/node-chassis-id"] = "2"
+			testNode.Annotations["k8s.ovn.org/l3-gateway-config"] = `{"default":{"mac-address":"0a:58:01:02:03:04","ip-addresses":["1.2.3.4/24"],"ip-address":"1.2.3.4/24","next-hops":["1.2.3.1"],"next-hop":"1.2.3.1"}}`
+			testNode.Annotations["k8s.ovn.org/node-transit-switch-port-ifaddr"] = fmt.Sprintf("{\"ipv4\":\"%s\"}", transitswitchSubnet)
+			testNode.Annotations["k8s.ovn.org/node-mgmt-port-mac-address"] = "0a:58:0a:01:01:02"
+			testNode.Annotations["k8s.ovn.org/zone-name"] = "foo"
+			testNode.Annotations["k8s.ovn.org/node-gateway-router-lrp-ifaddr"] = ""
+			fakeOvn.startWithDBSetup(dbSetup,
+				&v1.NamespaceList{
+					Items: []v1.Namespace{*newNamespace(config.Kubernetes.HostNetworkNamespace)},
+				},
+
+				&v1.NodeList{
+					Items: []v1.Node{testNode},
+				},
+			)
+
+			// Required to make sure a remote zone node is being added
+			fakeOvn.controller.zone = "node1"
+
+			gomega.Expect(fakeOvn.controller.WatchNamespaces()).To(gomega.Succeed(), "Namespace should be created fine")
+			err = fakeOvn.controller.WatchNodes()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			// Set annotations required to create remote zone node
```

Comment:
can't you do this in L1711 itself?
this makes sense when let's say the testNode used to be also in node1 zone and now you are changing it to foo zone but I am unsure why we are patching this annotation now and not at startup of the test.

Reply:
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/namespace.go#L312 adds management port and gateway router LRP IP to HostNetworkNamespace address set whenever the namespace gets created. I wanted node creation to sleep into retry loop until the namespace gets created and then I added required annotations to complete the node creation. 
I changed this logic now after discussing with you. Will push recent changes soon. Thanks for your help on this :)